From 4ddc87cecc6bf4e05ca89b511fdd1bb669a394e0 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 10 Apr 2017 23:21:32 +0100
Subject: [PATCH] configfile: Sync compatible flags from Clear Linux

This will help ensure we get the best defaults for our packages and still
allowing a relatively wide range of older systems to work fine (generic
tuning)

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 pisi/configfile.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/pisi/configfile.py b/pisi/configfile.py
index f3c9b8d..9b2968b 100644
--- a/pisi/configfile.py
+++ b/pisi/configfile.py
@@ -85,9 +85,9 @@ class BuildDefaults:
     jobs = "-j5"
     generateDebug = False
     enableSandbox = False # Dropping sandbox support soon
-    cflags = "-mtune=generic -march=x86-64 -ftree-vectorize -g2 -O2 -pipe -fPIC -Wformat -Wformat-security -fno-omit-frame-pointer -fstack-protector --param ssp-buffer-size=4 -fexceptions -D_FORTIFY_SOURCE=2 -feliminate-unused-debug-types -Wno-error -Wp,-D_REENTRANT"
-    cxxflags = "-mtune=generic -march=x86-64 -ftree-vectorize -g2 -O2 -pipe -fPIC -Wformat -Wformat-security -fno-omit-frame-pointer -fstack-protector --param ssp-buffer-size=4 -fexceptions -D_FORTIFY_SOURCE=2 -feliminate-unused-debug-types -Wno-error -Wp,-D_REENTRANT"
-    ldflags = "-Wl,--copy-dt-needed-entries -Wl,-O1 -Wl,-z,relro -Wl,-z,now"
+    cflags = "-mtune=generic -march=x86-64 -O2 -g -feliminate-unused-debug-types -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=32 -Wformat -Wformat-security -fasynchronous-unwind-tables -ftree-loop-vectorize -ftree-vectorize -fopt-info-vec -fno-omit-frame-pointer -Wno-error -Wp,-D_REENTRANT"
+    cxxflags = "-mtune=generic -march=x86-64 -O2 -g -feliminate-unused-debug-types -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-loop-vectorize -ftree-vectorize -fopt-info-vec -fno-omit-frame-pointer -Wno-error -Wp,-D_REENTRANT"
+    ldflags = "-Wl,--copy-dt-needed-entries -Wl,-z -Wl,now -Wl,-z -Wl,relro -Wl,-z,max-page-size=0x1000"
     buildhelper = "ccache"
     compressionlevel = 1
     fallback = "https://www.solus-project.com/source"
-- 
2.12.2

